pkg/bindinfo/utils.go

```diff
@@ -167,6 +170,110 @@ func readBindingsFromStorage(sPool util.DestroyableSessionPool, condition string
 	return
 }
 
+var (
+	// UpdateBindingUsageInfoBatchSize indicates the batch size when updating binding usage info to storage.
+	UpdateBindingUsageInfoBatchSize = 20
+	// MaxWriteInterval indicates the interval at which a write operation needs to be performed after a binding has not been read.
+	MaxWriteInterval = 6 * time.Hour
+)
+
+func updateBindingUsageInfoToStorage(sPool util.DestroyableSessionPool, bindings []*Binding) error {
+	toWrite := make([]*Binding, 0, UpdateBindingUsageInfoBatchSize)
+	for _, binding := range bindings {
+		lastSaved := binding.UsageInfo.LastSavedAt.Load()
+		if lastSaved == nil {
+			continue
+		}
+		lastUsed := binding.UsageInfo.LastUsedAt.Load()
+		if lastUsed == nil {
+			continue
+		}
+		intest.Assert(lastUsed.Compare(*lastSaved) >= 0, " lastUsed should be later than or equal to lastSaved")
+		if time.Since(*lastSaved) > MaxWriteInterval ||
+			// if the last used time is updated for more than 1 hour, we also write it back to storage.
+			time.Since(*lastUsed) > 1*time.Hour {
```

Comment:
How about only keeping the first condition here? Updating active bindings every 1h seems a little aggressive?